crates/turborepo-scm/src/manual.rs

```diff
@@ -9,13 +9,13 @@ use wax::{any, Glob, Pattern};
 
 use crate::{package_deps::GitHashes, Error};
 
-fn git_like_hash_file(path: &AbsoluteSystemPath, metadata: &Metadata) -> Result<String, Error> {
+fn git_like_hash_file(path: &AbsoluteSystemPath) -> Result<String, Error> {
     let mut hasher = Sha1::new();
     let mut f = path.open()?;
     let mut buffer = Vec::new();
-    f.read_to_end(&mut buffer)?;
+    let size = f.read_to_end(&mut buffer)?;
```

Comment:
`read_to_end` already attempts to do this optimally in terms of syscall count, we don't really need to help out.

`read_to_end` is this:
```rust
// Reserves space in the buffer based on the file size when available.
fn read_to_end(&mut self, buf: &mut Vec<u8>) -> io::Result<usize> {
    let size = buffer_capacity_required(self);
    buf.reserve(size.unwrap_or(0));
    io::default_read_to_end(self, buf, size)
}
```

`buffer_capacity_required` is this:
```rust
fn buffer_capacity_required(mut file: &File) -> Option<usize> {
    let size = file.metadata().map(|m| m.len()).ok()?;
    let pos = file.stream_position().ok()?;
    // Don't worry about `usize` overflow because reading will fail regardless
    // in that case.
    Some(size.saturating_sub(pos) as usize)
}
```